### PR TITLE
Migrate to http4s plugin

### PR DESCRIPTION
**Pull request by danicheg**
Towards resolving https://github.com/http4s/http4s-session/issues/152

### CODE REVIEW

**Comment by danicheg** [comment]
Perhaps that may take time to wait for Ross, so maybe we can call for help from @armanbilge with reviewing this. The diff is impressive (but not for Arman, I know 😆), but these changes were auto-generated mostly.

File: build.sbt [hunk 1]
```diff
@@ -87,71 +38,34 @@ lazy val core = project
...
-          Map("title" -> "code of conduct", "section" -> "code of conduct", "position" -> "100")
-        ),
-        file("LICENSE") -> ExtraMdFileConfig("license.md",
-                                             "page",
-                                             Map("title" -> "license", "section" -> "license", "position" -> "101")
-        )
-      )
-    )
-  }
+  .enablePlugins(TypelevelSitePlugin)
```
Comment (armanbilge) [inline build.sbt:68]:
```suggestion
  .enablePlugins(Http4sOrgSitePlugin)
```

File: build.sbt [hunk 2]
```diff
@@ -87,71 +38,34 @@ lazy val core = project
...
 
 // General Settings
 lazy val commonSettings = Seq(
```
Comment (armanbilge) [inline build.sbt:71]: Probably these can all be removed.

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+Session Management for http4s.
+
+## Quick Start
+
+To use http4s-session in an existing SBT project with Scala 2.13 or a later version, add the following dependencies to your
+`build.sbt` depending on your needs:
+
+```scala
+libraryDependencies ++= Seq(
+  "org.http4s" %% "http4s-session" % "<version>"
```
Comment (armanbilge) [inline docs/index.md:13]:
```suggestion
  "org.http4s" %% "http4s-session" % "@VERSION@"
```

**Review by armanbilge — APPROVED** [review]

File: docs/index.md (hunk 3 above)
Comment (danicheg) [reply docs/index.md]: That's interesting but `@VERSION@` uses a snapshot version. Do you know if we can bypass that behaviour?

Comment (danicheg) [reply docs/index.md]: Anyway, I'm not sure that this microsite will be published, so whatever, let's move on

**Comment by rossabaker** [comment]
Sorry, I'm in Day 3 (of 3) of laptops-closed meetings... thanks for moving this forward.

File: docs/index.md (hunk 3 above)
Comment (danicheg) [reply docs/index.md]: Hmm, that's bizarre that I saw the snapshot version locally, but at https://http4s.github.io/http4s-session/ we are good!

**Pull request merged**